webserver/pkg/infrastructure/mqtt/device_monitor.go

```diff
@@ -0,0 +1,218 @@
+		default:
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to subscribe to sensor: %w", err)
+	}
+
+	if err := dm.client.SubscribeToInfo1(func(data Info1Data) error {
+		select {
+		case info1Chan <- data:
+		default:
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to subscribe to INFO1: %w", err)
+	}
+
+	if err := dm.client.SubscribeToInfo2(func(data Info2Data) error {
+		select {
+		case info2Chan <- data:
+		default:
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to subscribe to INFO2: %w", err)
+	}
+
+	if err := dm.client.SubscribeToLWT(func(status string) error {
+		select {
+		case lwtChan <- status:
+		default:
+		}
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to subscribe to LWT: %w", err)
+	}
+
+	if err := dm.client.RestartDevice(); err != nil {
+		return fmt.Errorf("failed to send restart command: %w", err)
+	}
+
+	time.Sleep(3 * time.Second)
+
+	go dm.monitorLoop(sensorChan, info1Chan, info2Chan, lwtChan)
+
+	return nil
+}
+
+func (dm *DeviceMonitor) monitorLoop(sensorChan <-chan SensorData, info1Chan <-chan Info1Data, info2Chan <-chan Info2Data, lwtChan <-chan string) {
+	for {
+		select {
+		case <-dm.ctx.Done():
+			return
+		case sensorData := <-sensorChan:
+			dm.handleSensorData(sensorData)
+		case info1Data := <-info1Chan:
+			dm.handleInfo1Data(info1Data)
+		case info2Data := <-info2Chan:
+			dm.handleInfo2Data(info2Data)
+		case lwtStatus := <-lwtChan:
+			dm.handleLWTStatus(lwtStatus)
+		}
+	}
+}
+
+func (dm *DeviceMonitor) handleSensorData(data SensorData) {
+	timestamp, err := time.Parse("2006-01-02T15:04:05", data.Time)
+	if err != nil {
+		timestamp = time.Now()
+	}
+
+	dm.mu.Lock()
+	dm.status.UpdatePower(data.ENERGY.Power, timestamp)
+	dm.status.UpdateVoltage(data.ENERGY.Voltage)
+	statusCopy := dm.status
+	dm.mu.Unlock()
+
+	dm.notify(statusCopy)
+}
+
+func (dm *DeviceMonitor) handleInfo1Data(data Info1Data) {
+	dm.mu.Lock()
+	dm.status.UpdateInfo1(data.Info1.Version, data.Info1.Module)
+	statusCopy := dm.status
+	dm.mu.Unlock()
+
+	dm.notify(statusCopy)
+}
+
+func (dm *DeviceMonitor) handleInfo2Data(data Info2Data) {
+	dm.mu.Lock()
+	dm.status.UpdateInfo2(data.Info2.Hostname, data.Info2.IPAddress)
+	statusCopy := dm.status
+	dm.mu.Unlock()
+
+	dm.notify(statusCopy)
+}
+
+func (dm *DeviceMonitor) handleLWTStatus(status string) {
+	dm.mu.Lock()
+	dm.status.UpdateLWTStatus(status)
+	statusCopy := dm.status
+	dm.mu.Unlock()
+
+	dm.notify(statusCopy)
+}
+
+func (dm *DeviceMonitor) notify(status *domain.DeviceStatus) {
+	dm.subscribersMu.RLock()
+	subscribers := make([]func(*domain.DeviceStatus), len(dm.subscribers))
+	copy(subscribers, dm.subscribers)
+	dm.subscribersMu.RUnlock()
+
+	for _, callback := range subscribers {
+		callback(status)
+	}
+}
+
+func (dm *DeviceMonitor) PowerOn() error {
+	return dm.client.PublishPowerCommand(true)
+}
+
+func (dm *DeviceMonitor) PowerOff() error {
+	return dm.client.PublishPowerCommand(false)
```

Comment:
TODO: add ticket for functionality, intended to be used from internal console with auth